doc/sphinx/addendum/type-classes.rst

```diff
@@ -322,12 +340,32 @@ Summary of the commands
    Like any command declaring a record, this command supports the
    :attr:`universes(polymorphic)`, :attr:`universes(template)`,
    :attr:`universes(cumulative)`, and :attr:`private(matching)`
-   attributes.
+   attributes. In addition, it supports the :attr:`modes` attribute.
+
+   .. attr:: modes
+
+      Sets the resolution modes of the class at declaration time,
+      using the same syntax as :cmd:`Hint Mode`: :n:`{+, {+ {| + | ! | - } }}`, where
+      multiple modes can be declared, separated by a comma.
```

Comment:
```suggestion
   .. attr:: modes "{+ {| + | ! | - } }"

      Sets the resolution modes of the class at declaration time,
      using the same markers as in :cmd:`Hint Mode`.
```

It looks like you support multiple modes in a single string.  Allowing a comma-separated list seems unnecessary.

Reply:
A mode declaration is a space-separated list of `+ | ! | -`, so we need a comma to separate multiple such lists

Reply:
`modes "+!"` is either not supported or not equivalent to `modes "+", "!"`?  Does the order of the items make any difference?  And repetitions are pointless but harmless, e.g. "++"?

Please get the right syntax on the `.. attr::` line, though.

Reply:
No, they are not equivalent. `modes="+, !"` sets two different modes (input and head-not-evar) for a single index typeclass. `modes="+ !"` sets a single mode declaration for a class with two indices. `modes="+ -, - +"` sets two mode declarations for a class with two indices. Repetitions like `modes="!","!"` are useless yes, as `,` models a disjunction, and order does not matter between comma-separated declarations, but of course it does if you have two indices.

Reply:
If the spaces are required in `modes="+ -, - +"`, that should be mentioned.

I suppose the number of indices is the number of parameters for the class?
If so, might be nice to mention here that the number of parameters should match the class.  Maybe even link to http://localhost:63342/coq2/doc/sphinx/_build/html/addendum/type-classes.html#class-and-instance-declarations.

Reply:
No, spaces are not significant. I'll add something about matching mode declarations with the parameters

Reply:
We are already in the section of the manual that deals with class declarations, so a link looks unnecessary here.